scheduler/src/cook/mesos/api.clj

```diff
@@ -1291,6 +1296,33 @@
       true
       [false {::error "You are not authorized to create jobs"}])))
 
+(defn no-job-exceeds-quota?
+  "Check if any of the submitted jobs exceed the user's total quota,
+   in which case the job would never be schedulable (unless the quota is increased).
+
+   If none of the jobs individually seem to exceed the user's quota, returns true;
+   otherwise, the following error structure is returned:
```

Comment:
If the jobs are not grouped, should we only block the individual jobs that exceed quota?

Reply:
I thought about that, but I concluded that the return status would be too messy in that case. Either we have to do something like a multi-status return code (HTTP 207 Multi-Status), or return a status that's misleading about part of the request (either we say it failed when some jobs *were* successfully submitted, or we say it was successful when in fact some jobs were rejected). Simply rejecting the whole request outright seemed significantly cleaner than either of those options.

Reply:
Makes sense